### PR TITLE
Support image with multiple tags

### PR DESCRIPTION
**Pull request by RoryH**
Nova assumed that each image would have the tag it was looking for as the first tag. If you happened to have the same image with multiple tags this would break

### CODE REVIEW

File: nova/core/managers/docker_manager.py [hunk 1]
```diff
@@ -32,17 +32,17 @@ def images(self):
         return self.docker_client.images()
 
     def find_image(self, looking_for_tag):
-        matching_images = [d for d in self.docker_client.images() if d['RepoTags'] is not None and d['RepoTags'][0].endswith(looking_for_tag)]
+        matching_images = [d for d in self.docker_client.images() if d['RepoTags'] is not None and looking_for_tag in d['RepoTags']]
```
Comment (gheine) [inline nova/core/managers/docker_manager.py]: 'in' doesn't sound like a full replacement for 'endswith'?

Comment (RoryH) [reply nova/core/managers/docker_manager.py]: true, but as I understand it was trying to match the image name and version in the format `docker-image-name:1.2.3`, so I'm not sure of a use case where you would want .endswith()

I can refine it to do the same logic if we believe that's what is intended.

Comment (gheine) [reply nova/core/managers/docker_manager.py]: No I guess this makes sense. Just verified that what's passed in as looking_for_tag contains the service name and not just the version.

Comment (nicodom) [reply nova/core/managers/docker_manager.py]: `looking_for_tag = "%s:%s" % (self._service_manager.service_name, self._build_id)`
@RoryH is right.

Comment (nicodom) [reply nova/core/managers/docker_manager.py]: Sorry @gheine didn't get the update with your comment.

**Pull request merged**